lntest/itest/lnd_channel_graph_test.go

```diff
@@ -775,3 +776,194 @@ func testNodeAnnouncement(net *lntest.NetworkHarness, t *harnessTest) {
 	ctxt, _ = context.WithTimeout(ctxb, channelCloseTimeout)
 	closeChannelAndAssert(ctxt, t, net, net.Bob, chanPoint, false)
 }
+
+// testUpdateNodeAnnouncement ensures that the RPC endpoint validates
+// the requests correctly and that the new node announcement is brodcasted
+// with the right information after updating our node
+func testUpdateNodeAnnouncement(net *lntest.NetworkHarness, t *harnessTest) {
+	ctxb := context.Background()
+
+	// Launch notification clients for alice, such that we can
+	// get notified when there are updates in the graph.
+	aliceSub := subscribeGraphNotifications(ctxb, t, net.Alice)
+	defer close(aliceSub.quit)
+
+	// Use the default address *explicitly* to make asserts more
+	// readable
+	advertisedAddrs := []string{
+		"127.0.0.1:5569",
+	}
+
+	var lndArgs []string
+	for _, addr := range advertisedAddrs {
+		lndArgs = append(lndArgs, "--externalip="+addr)
+	}
+
+	dave := net.NewNode(t.t, "Dave", lndArgs)
+	defer shutdownAndAssert(net, t, dave)
+
+	// Get dave default information so we can compare
+	// it lately with the new brodcasted updates
+	nodeInfoReq := &lnrpc.GetInfoRequest{}
+	resp, err := dave.GetInfo(ctxb, nodeInfoReq)
+	if err != nil {
+		t.Fatalf("unable to get dave's information: %v", err)
+	}
+
+	defaultDaveNodeAnn := &lnrpc.NodeUpdate{
+		Alias: resp.Alias,
+		Color: resp.Color,
+		NodeAddresses: []*lnrpc.NodeAddress{
+			{Addr: advertisedAddrs[0], Network: "tcp"},
+		},
+	}
+
+	// Test RPC param validation for each one of the fields
+	invalidNodeAnnReq := &lnrpc.NodeAnnouncementUpdateRequest{
+		Color: "invalidValue",
+	}
+
+	ctxt, _ := context.WithTimeout(ctxb, defaultTimeout)
+	_, err = dave.UpdateNodeAnnouncement(ctxt, invalidNodeAnnReq)
+	if err == nil {
+		t.Fatalf("faild to validate an invalid color for an update node announcement request: %v", err)
+	}
+
+	// We cannont differentiate between requests with Alias = "" and requests
+	// that do not provide that field. If a user sets Alias = "" in the resquest
+	// the field will simply be ignored. The request must fail because no
+	// modifiers are applied
+	invalidNodeAnnReq = &lnrpc.NodeAnnouncementUpdateRequest{
+		Alias: "",
+	}
+	ctxt, _ = context.WithTimeout(ctxb, defaultTimeout)
+	_, err = dave.UpdateNodeAnnouncement(ctxt, invalidNodeAnnReq)
+	if err == nil {
```

Comment:
We've started to use the `require` package in the itests (and really any tests) to cut down on the line noise that assertions add: https://pkg.go.dev/github.com/stretchr/testify/require

Reply:
perfect, I will use that way then